opentelemetry-sdk/src/metrics/meter.rs

```diff
@@ -485,6 +519,75 @@ impl InstrumentProvider for Meter {
     }
 }
 
+/// Validation policy for instrument
+#[derive(Clone, Copy)]
+enum InstValidationPolicy {
+    HandleGlobalAndIgnore,
+    /// Currently only for test
+    #[cfg(test)]
+    Strict,
+}
+
+fn validate_instrument_config(
+    name: &str,
+    unit: Option<&Unit>,
+    policy: InstValidationPolicy,
+) -> Result<()> {
+    match validate_instrument_name(name).and_then(|_| validate_instrument_unit(unit)) {
+        Ok(_) => Ok(()),
+        Err(err) => match policy {
+            InstValidationPolicy::HandleGlobalAndIgnore => {
+                global::handle_error(err);
+                Ok(())
+            }
+            #[cfg(test)]
+            InstValidationPolicy::Strict => Err(err),
+        },
+    }
+}
+
+fn validate_instrument_name(name: &str) -> Result<()> {
+    if name.is_empty() {
+        return Err(MetricsError::InvalidInstrumentConfiguration(
+            INSTRUMENT_NAME_EMPTY,
+        ));
+    }
+    if name.len() > 63 {
```

Comment:
Do you plan to handle part of fix for #1265 to increase the char limit to 255 separately ?

Reply:
@lalitb thanks for review, I'll work on this PR for char limit to 255 !